src/analysis/state_helper.rs

```diff
@@ -63,12 +63,21 @@ fn find_previous_non_empty_split_and_comparison_time(
 /// Calculates the comparison's segment time of the segment with the timing
 /// method specified. This is not calculating the current attempt's segment
 /// times.
+///
+/// # Panics
+///
+/// Panics if the provided `segment_index` is greater than or equal to
+/// `run.len()`.
 pub fn comparison_segment_time(
```

Comment:
I don't think that the current implementation of this method makes sense.

- This provides a combined segment time, which means that the detailed timer's logic will change when switching over to it.
- Using the best segment time only makes sense if we're using non-combined segment times (because, by definition, the best segment time is a non-combined segment time). That's why the detailed timer was using it previously, but the splits component was not.

Reply:
Yeah, this wasn't meant to be finished.

Reply:
I would prefer if there were two separate methods in `state_helper`: one for combined segment times and one for non-combined segment times. The non-combined segment time method would do the best segment time override, while the combined segment time method would not. The splits component would use the combined segment time method, and the detailed timer component would use the non-combined segment time method. The segment time component would use whichever one we think makes the most sense (which is the discussion we just had in the other thread that didn't reach a clear answer).

Reply:
So if we end up splitting comparison segment time into two distinct methods and splits keep using the combined one, then there's no way for the splits to show the best segment times as a column. Are we sure that's what we want? Is there a disadvantage to just switch to the best segment times despite us usually showing combined segment times? The problem is that just like best split times and co. we can't really derive any segment times from it, as the times aren't truly building on top of each other.

Reply:
Personally I'd prefer just having a single method and explicitly handling empty segments in the detailed timer itself if the resulting code is cleaner than having 2 different methods with almost identical code.

Reply:
Eh, I guess it's fine to use the best segment times then, even if it's a little inconsistent.

Reply:
Looks like we might actually have to have 2 different functions.

The cleanest way I can think of handling things in the detailed timer is to use a wrapper function and at that point we might as well just have 2 functions.

Reply:
In the end I went with the solution of having 2 different methods. I couldn't find a way to cleanly keep the current behavior of the detailed timer besides creating a wrapper function.